### PR TITLE
 Setting O_DIRECT flag in Integration tests

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Currently we are using linux command to clear the cache, instead we should replace with the O_DIRECT flag while reading any file.
os.OpenFile(name, syscall.O_DIRECT)

### Testing
1. Manual - checked gcsfuse.log. not reading from kernal. Integration tests are working fine for both the cases(testbucket and MountDir)
2. UnitTest - N/A
3. IntegrationTest - N/A

### CODE REVIEW

File: tools/integration_tests/implicitdir/file_operations_test.go [hunk 1]
```diff
@@ -27,15 +28,19 @@ import (
 
 func TestRenameFile(t *testing.T) {
 	fileName := setup.CreateTempFile()
-	err := setup.ClearKernelCache()
+
+	file, err := os.OpenFile(fileName, syscall.O_DIRECT, 0666)
```
Comment (raj-prince) [inline tools/integration_tests/implicitdir/file_operations_test.go:32]: Please make a const variable for 0666 and use it everywhere.

**Review by raj-prince — COMMENTED** [review]
Minor comments.

File: tools/integration_tests/implicitdir/file_operations_test.go (hunk 1 above)
Comment (Tulsishah) [reply tools/integration_tests/implicitdir/file_operations_test.go]: fixed

File: tools/integration_tests/implicitdir/write_test.go [hunk 2]
```diff
@@ -17,14 +17,15 @@ package implicitdir_test
...
+	"syscall"
 	"testing"
 
 	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/setup"
 )
 
 func TestWriteAtEndOfFile(t *testing.T) {
 	fileName := setup.CreateTempFile()
-	f, err := os.OpenFile(fileName, os.O_APPEND|os.O_WRONLY, 0600)
+	f, err := os.OpenFile(fileName, os.O_APPEND|os.O_WRONLY|syscall.O_DIRECT, 0600)
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply tools/integration_tests/implicitdir/write_test.go]: fixed

File: tools/integration_tests/implicitdir/file_operations_test.go [hunk 3]
```diff
@@ -27,15 +28,19 @@ import (
...
 	if err != nil {
-		t.Errorf("Clear Kernel Cache: %v", err)
+		t.Errorf("Error in the opening file: %v", err)
 	}
+	defer file.Close()
 
-	content, err := os.ReadFile(fileName)
+	content := make([]byte, 100)
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply tools/integration_tests/implicitdir/file_operations_test.go]: fixed

**Comment by Tulsishah** [comment]
> Minor comments.

Fixed

File: tools/integration_tests/implicitdir/write_test.go [hunk 4]
```diff
@@ -39,7 +40,7 @@ func TestWriteAtEndOfFile(t *testing.T) {
 
 func TestWriteAtStartOfFile(t *testing.T) {
 	fileName := setup.CreateTempFile()
-	f, err := os.OpenFile(fileName, os.O_WRONLY, 0600)
+	f, err := os.OpenFile(fileName, os.O_WRONLY|syscall.O_DIRECT, 0600)
```
Comment (raj-prince) [inline tools/integration_tests/implicitdir/write_test.go:43]: Replace here also.

Comment (Tulsishah) [reply tools/integration_tests/implicitdir/write_test.go]: fixed

**Review by raj-prince — APPROVED** [review]

**Pull request merged**